dynamic/clientset/clientset.go

```diff
@@ -32,78 +32,93 @@ import (
 	dynamicobject "openebs.io/metac/dynamic/object"
 )
 
-// Clientset manages various k8s client related operations against one
-// or more discovered API resource(s). Clientset has the ability to
-// provide dynamic client for specific resource.
+// Clientset provides dynamic client(s) corresponding
+// to discovered API resource(s)
 type Clientset struct {
-	config          rest.Config
-	resourceManager *dynamicdiscovery.APIResourceManager
-	dynamicClient   dynamic.Interface
+	config           rest.Config
+	discoveryManager *dynamicdiscovery.APIResourceDiscovery
+	dynamicClient    dynamic.Interface
 }
 
 // New returns a new instance of Clientset
 func New(
-	config *rest.Config, resourceMgr *dynamicdiscovery.APIResourceManager,
+	config *rest.Config,
+	resourceMgr *dynamicdiscovery.APIResourceDiscovery,
 ) (*Clientset, error) {
-
 	dc, err := dynamic.NewForConfig(config)
 	if err != nil {
-		return nil, errors.Wrapf(err, "New clientset failed")
+		return nil, errors.Wrapf(
+			err,
+			"Failed to create dynamic clientset",
+		)
 	}
-
 	return &Clientset{
-		config:          *config,
-		resourceManager: resourceMgr,
-		dynamicClient:   dc,
+		config:           *config,
+		discoveryManager: resourceMgr,
+		dynamicClient:    dc,
 	}, nil
 }
 
-// HasSynced flags if resources managed by this clientset have
-// synced i.e. refreshed from the API server
+// HasSynced returns true if all the discovered resources
+// are synced _i.e. are refreshed from the API server_
 func (cs *Clientset) HasSynced() bool {
-	return cs.resourceManager.HasSynced()
+	return cs.discoveryManager.HasSynced()
 }
 
-// GetClientByResource returns the resource client corresponding to
-// the given version & resource name (i.e. plural of kind)
-func (cs *Clientset) GetClientByResource(apiVersion, resource string) (*ResourceClient, error) {
-	// Look up the requested resource in discovery.
-	apiResource := cs.resourceManager.GetByResource(apiVersion, resource)
+// GetClientForAPIVersionResource returns the dynamic client for
+// the given apiversion & resource name _(i.e. plural of kind)_
+func (cs *Clientset) GetClientForAPIVersionResource(
+	apiVersion string,
+	resource string,
+) (*ResourceClient, error) {
+	// look up the requested resource from discovered resources
+	apiResource := cs.discoveryManager.GetAPIForAPIVersionAndResource(
+		apiVersion,
+		resource,
+	)
 	if apiResource == nil {
 		return nil, errors.Errorf(
-			"Failed to initialise dynamic client for resource %q in apiVersion %q",
+			"Can't get client for %q with api version %q: Resource is not discovered",
```

Comment:
nit: `Failed to get client...` would be more consistent with the rest of the error messages and make it easier to grep the logs.

Reply:
thanks @luisdavim I will fix it